lib/components/map/route-viewer-overlay.js

```diff
@@ -15,10 +13,27 @@ class RouteViewerOverlay extends MapLayer {
   componentWillUnmount () {}
 
   componentWillReceiveProps (nextProps) {
+    // helper fn to check if geometry has been populated for all patterns in route
+    const isGeomComplete = routeData => {
+      return (
+        routeData &&
+        routeData.patterns &&
+        Object.values(routeData.patterns).reduce(
```

Comment:
You could use [Array.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) for this.